Freezeman

ehenrion opened the pull request.

Commit:
First draft for freezeman integration to the monitor

Commit:
Fixing the email tempaltes

Commit:
Merge branch 'main' of github.com:c3g/monitor into freezeman

Conflicts:
	workflows/mgi/monitor.nf
	workflows/testing.nf

Commit:
Adding path for freezeman run info files to watch

Commit:
Updating outdir for debug profile

Commit:
Fixed typo in config

Commit:
updating path for testing assets

Commit:
Setting monitor for Freezeman LIMS !

Commit:
Freezeman monitor - Illumina runs and MGI runs are now all watched

garantjm committed:
garantjm dev branch, child process Mem issue

First attempt at a correction of the monitor memory overuse issue. Hypothesis: Stopping the monitor desn't kill the child processes and the stop'n'launch of debugging increases the amount of loose child processes that hug the memory.

Commit:
Updating the email start groovy template

Commit:
Email templates fixes and updates

Commit:
MGI T7 - fixed used of flag files path from config

Commit:
Clarity - fixing use of config for MGI T7 flag files

Commit:
Start working on freezeman ingestion

garantjm committed:
Genpipes as submodules

The monitor calls a process to clone or link genpipes for each channel instances, it shouldn't even be a process

garantjm committed:
Merge branch 'fzmn-child-process' into freezeman

Conflicts:
	workflows/freezeman/monitor.nf

Commit:
Updating the freezemannIngestor script and its call bty the momitor

Commit:
Merge branch 'freezeman' of https://github.com/c3g/monitor into freezeman

Commit:
fix config

Commit:
Fix run dir for illumina runs

garantjm committed:
Readme, .py arguments format

Include a first version of a README, modification of arguments in freezemanIngestor.py to a more common format

garantjm committed:
Merge branch 'freezeman' into fzmn-child-process

Commit:
Fix debug.db path in config

garantjm committed:
Revision of argparser, include password prompt

Positional argument for JSON, inclusion of metavar, option to have password prompt and redirection of else error message to stderr

garantjm committed:
Changed GenPipes submodule URL to a public accessible https

Commit:
Fixing ini for genpipes as well as email start template

Commit:
Changing definition of custom_ini

Commit:
Fixes on emails & BeginRun

Commit:
Resetting 'onstart' emails for debug profile

Commit:
Merge branch 'main' into freezeman

Conflicts:
	assets/email_run_finish.groovy
	assets/production.t7.ini
	lib/Eventfile.groovy
	lib/MetadataDB.groovy
	nextflow.config
	workflows/clarity/monitor.nf
	workflows/mgi/launch.nf

Commit:
Updated .gitmodules with specific 'run_processing' branch from genpipes

Commit:
Add correct imports in clarity/launch.nf

garantjm committed:
Update doc information

Notes added after meeting with the data team

garantjm committed:
GetGenpipes process removed

Commit:
few adkjustmens in config + other minor changes

garantjm commented:
Monitor modification in order to manage the run processing of samples that are submitted and managed via Freezeman. This work is preparing the monitor to be the sole run processing management launcher in order to put Freezeman in production in the lab.

garantjm committed:
Merge branch 'freezeman' into fzmn-child-process

Commit:
Strengthen the parsing of the MultiQC json

Commit:
Corrected `outdir` in BeginRun

garantjm committed:
Merge branch 'freezeman' into fzmn-child-process

garantjm committed:
Add fzmn-child-process configs & ini

Specific values allows to run tests under non-freezeman users

garantjm committed:
GenPipes submodules updated

There was a rebase of run_processing branch to include changes coming from the dev branch

garantjm committed:
GetGenpipes leftovers removed, validation report ssh  key, more logs

Remaining code of GetGenpipes removed, arbutus_uds_id_rsa ssh key added as parameter to upload validation reports, increased logs to investigate launch delay

garantjm committed:
Merge branch 'fzmn-child-process' into freezeman

Merge excluding fzmn-child-process specific files

garantjm committed:
Add key arg to summary upload & errorStrategy ignore for emails

Genap uploads work with the ney ssh key parameter, it's now added to summary upload. Email failures won't stop the monitor.

garantjm committed:
FreezemanIngestor connect & auth fixed, new arguments

FreezemanIngestor is more flexible to run in or out of Abacus, multiple changes in the argparse set-up not reflected in the nextfolw process yet

garantjm committed:
Freezeman runinfo file modifications

Changes in the fields that are required to import the updated runinfo files and send the validation json to ingest

garantjm committed:
GenPipes update for run validation json, error strategy handling, documentation

Genpipes update to include fields needed for Freezeman ingestion. Error strategy retries most steps once and keeps the monitor running afterward. README includes some Abacus specific set-up that are required for the monitor to run successfully.

garantjm committed:
Clarity BeginRun fixed with new GenPipes submodule

Clarity BeginRun could not launch now that GenPipes is an updated submodule

garantjm committed:
Path changes towards dedicated freezeman-processing

garantjm committed:
GenPipes RunProcessing fixes for T7

Importantly it fixes the paths issues when ContainerName is not equal to RunCounter. It is still missing the MGI run report and a fix for dual second barcode in the R2 read for dual barcodes libraries.

garantjm committed:
Update MultiQC module

garantjm committed:
Genome null length issue and tweaks for freezeman-qc

garantjm committed:
Genomic database copy paste fix in genpipes

garantjm committed:
Duplicate genomic database fix in genpipes readset

garantjm committed:
Ingestion in Freezeman

Ingestion is done using the techdevadmin user on either qc or prod instances via the profile used

garantjm committed:
FreezemanIngest running, freezemanIngestor sending multiple Validation report at once, workDir in scratch space

garantjm committed:
Commenting out support for HiseqX, Iseq1 & Iseq2

To speed up launch, reduce & restrain the watchpaths and move the sqliteDB to start fresh.

garantjm committed:
Add project_name to validation report for Freezeman ingestion and exit code to ingestor

garantjm committed:
NovaseqX paths & channels added, genpipes supported

garantjm committed:
New MultiQC run validation reports directory on genap's datahub

garantjm committed:
Reorder to have NovaseqX ready up before Novaseq

Only for convenience of testing faster NovaseqX

garantjm committed:
NovaseqX flowcell string has a different position in path

4th from left is more appropriate than second from right because the project is not in path. A similar solution might apply to Miseq paths

garantjm committed:
Run_processing relies on GenPipes version of adapter_settings_format.txt

garantjm committed:
Repair profiles string in config

garantjm committed:
Update MulitQC, add year to outdir path & clean logs

garantjm committed:
NovaseqX index job, WGBS not aligned, summaryReport T7

garantjm committed:
Add year to monitor.nf watchPath

garantjm committed:
HotFix: params.outdir set instead of .mgi.outdir

garantjm committed:
Genpipes run_validation report with url towards datahub Freezeman_validation folder

garantjm committed:
Fix validation report URL in finish email template, update onfinish email list

garantjm committed:
GenPipes copy step includes a compression job of logs.

garantjm committed:
Update mugqic_tools to fix types in alignment metrics for run validation

garantjm committed:
Folder name added to finish email

Requested by Janick St-Cyr for her convenience

garantjm committed:
Replace run value from MultiQC with run_name from run info file, add the start email table

garantjm committed:
Simple copy of each MultiQC run to the run processing folder

garantjm committed:
Includes QoL changes requested by the lab

Includes a fix for the tar compression to ignore its own outputs and removing the MultiQC .html from email attachments

garantjm committed:
Sex concordance fix and new certificates

garantjm committed:
Remove retries for BeginRun to make sure there are no double launches on qsub failures

garantjm committed:
Update MultiQC_C3G to v1.23

garantjm committed:
Update indexes, barcodes, adapters in GenPipes

garantjm committed:
Add sortmerna for RNAseq reports

garantjm committed:
Removing retries to reduce the potential issues associated

garantjm committed:
Merge sortmerna & copy step PR in GenPipes

garantjm committed:
Genpipes fix for Danielle missing mkdir in fastq move

garantjm committed:
Change the glob expression that ignores dnbseqt7 for a glob restricted to novaseqs sequencers

garantjm committed:
Adding Ioannis Ragoussis to emailonfinish list

garantjm committed:
MultiQC reports rsynced towards /lb/robot/ as well as /nb/Research

MareikeJaniak committed:
update genpipes version within monitor

throttling fixes, bam names unique

MareikeJaniak committed:
add curl command to send data to dashrunr after multiqc report is generated

MareikeJaniak committed:
update genpipes version within monitor

MareikeJaniak committed:
update multiqc version, move multiqc report sync and dashrunnr update elsewhere

MareikeJaniak committed:
debug FinalSync

MareikeJaniak committed:
debug final sync

MareikeJaniak committed:
debug final sync

MareikeJaniak committed:
revert FinalSync for now, not working

MareikeJaniak committed:
add FinalSync to sync nb and lb, push to run dashboard

MareikeJaniak committed:
moving FinalSync to happen after RunMultiQC

MareikeJaniak committed:
update multiqc module version

MareikeJaniak committed:
debug final sync for monitor, fix genap link in finish email template

MareikeJaniak committed:
debug final sync for monitor

MareikeJaniak committed:
debug final sync, format analysis dir

MareikeJaniak committed:
debug final sync, escaping

MareikeJaniak committed:
debug final sync, remove curl to dashrunnr for now

MareikeJaniak committed:
Merge pull request #3 from c3g/freezeman_finalSync

add FinalSync to sync nb and lb, push to run dashboard

MareikeJaniak committed:
access year from runinfo, instead of using wild card for FinalSync

MareikeJaniak committed:
import analysis dir as element of list instead of whole list, update FinalSync and finish template accordingly

MareikeJaniak committed:
clean up FinalSync for PR

MareikeJaniak committed:
Merge pull request #4 from c3g/freezeman_finalSync

Freezeman final sync

MareikeJaniak committed:
update genpipes version within monitor

MareikeJaniak committed:
update genpipes version within monitor - copy step fix

MareikeJaniak committed:
update genpipes version within monitor

MareikeJaniak committed:
update multiqc version in monitor

MareikeJaniak committed:
update genpipes version within monitor

garantjm committed:
Update GenPipes version within monitor

MareikeJaniak committed:
update genpipes version within monitor

MareikeJaniak committed:
update genpipes version within monitor

garantjm committed:
Update MultiQC undetermined barcodes and GenPipes checkpoints

garantjm committed:
Walltime for MultiQC and GenPipes update

garantjm committed:
Cleanup true, Datahub local alternative

garantjm committed:
Fix symlinks towards local reports

garantjm committed:
MultiQC data copy towards /lb/robot/research at checkpoints in parallel of GenapUpload

garantjm committed:
Fix MiSeq support

garantjm committed:
Associated GenPipes fix for Miseq

garantjm committed:
Remove GenapUpload to sftp, change the process name to UpdateReport

garantjm committed:
MultiQC update

garantjm committed:
Update EOF overlooked

garantjm committed:
Update GenPipes to run_processing : fixes for lanes with dual and single indexes

garantjm committed:
Removing MiSeq from BeginRun & EmailAlertStart

garantjm committed:
Removing MiSeq from RTAComplete.txt watchChannels

garantjm committed:
Lazy fix for exit status 1 after the softlink filecheck of the report link

garantjm committed:
Using launch workflow without BeginRun and EmailAlertStart

garantjm committed:
Update GenPipes uniq filename, adapt 2026
